deploy/docker/fs/opt/appsmith/infra_details.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -e
+
+export INFRA_FILE="$TMP/infra.json"
+export MOUNT_PATH="/appsmith-stacks"
+
+## Check for the cloud provider
+function get_cloud_provider() {
+    release_details=$(uname -r)
+    if [[ $release_details == *"amzn"* ]];then
+       cloud_provider="amazon";
+    elif [[ $release_details == *"azure"* ]];then
+       cloud_provider="azure";
+    elif [[ $release_details == *"cloud"* ]];then
+       cloud_provider="gcp";
+    else
+       cloud_provider="local";
+    fi
+}
+
+## Get if it is deployed on docker or kubernetes
+function get_tool() {
+    if [[ -z "${KUBERNETES_SERVICE_HOST}" ]]; then
+        dep_tool="docker";
+    else
+        dep_tool="kubernetes";
+    fi
+}
+
+## Check if EFS is mounted
+function check_for_efs() {
+## TOOO LAME NEED TO CHANGE
+if findmnt --mountpoint $MOUNT_PATH | grep -q nfs; then
+   efs="present"
+else
+   efs="absent"
+fi    
+}
+
+## Check the hostname and IP
+function get_username_ip() {
+    hostname=$(cat /etc/hostname)
```

Comment:
Double quote _all_ subshell-outputs and variable usages.

```suggestion
    hostname="$(cat /etc/hostname)"
```